doc/recognition.xml

```diff
@@ -336,8 +336,11 @@ set this to <K>false</K>. The result is either <K>fail</K> or
 <K>false</K> if the method fails or a recognition info record <C>ri</C>.
 If the method created a leaf then <C>ri</C> will be a leaf, otherwise
 it will have the attribute <Ref Attr="Homom"/> set, but no factor or
-kernel have been created or recognised yet. You can use for example
-the methods in <Ref Var="FindHomMethodsPerm"/> or
+kernel have been created or recognised yet. Optionally,
+<Ref Attr="HomomSourceCheck"/> can be set. When this is set, <Ref Attr="Homom"/>
+can fail, error, or produce incorrect answers for any value for which
+<Ref Attr="HomomSourceCheck"/> returns false.
+You can use for example the methods in <Ref Var="FindHomMethodsPerm"/> or
```

Comment:
I really think this needs more explanation as to *why* this matters. From just reading this, it is not clear to me.

Naively I would say: "why can't the Homom homomorphism just return `fail`, and be done with it? I am guessing the answer is: "because some of them are stock GAP homomorphisms, and they produces errors.". Though perhaps it is also "... and they produce nonsense results for nonsense inputs", which arguable would be worse.

That said, of course a second strategy is possible: We can just demand that morphisms must always validate their input. If we were to do that, then we wouldn't need this extra function, at the cost of making it slightly more annoying to make homomorphism objects -- though we could provide a helper function to make it very easy.

BTW, note that for `ActionHomomorphism`, one can set the `actioncanfail` option to get `fail` instead of errors -- but that requires one to pass that to every call of `ImageElm`.. Hrm.

Reply:
From experience this week, built in GAP homomorphisms can Error, enter the break loop, or produce incorrect answers, if given a value not in their source.

It seems like GAP's general attitude (and documentation) is that this is how homomorphisms work, so I feel like that's the more natural thing to do (and, will make it easier to disable these checks late, if/when we want). It does of course mean we need to remember to call this extra function where required.